blog/_src/posts/2022-06-21-shallow-typed-racket.scrbl

```diff
@@ -0,0 +1,335 @@
+#lang scribble/manual
+
+Title: Shallow and Optional Typed Racket
+Date: 2022-06-21T23:43:38
+Tags: Typed Racket
+
+@(require
+   scribble/example
+   (only-in scribble/eval interaction-eval interaction/no-prompt)
+   (for-label typed/racket/base))
+
+@(define deep-eval (make-base-eval #:lang 'typed/racket))
+@(define shallow-eval (make-base-eval #:lang 'typed/racket/shallow))
+@(define optional-eval (make-base-eval #:lang 'typed/racket/optional))
+
+@(define (tech/reference . text)
+   (keyword-apply tech '(#:doc) '((lib "scribblings/reference/reference.scrbl")) text))
+
+@(define-syntax-rule (module-example #:eval ev #:label lbl lang-datum mod-code ... ex-code)
+   (list
+     (if lbl (list lbl) '())
+     (racketmod lang-datum mod-code ...)
+     (interaction-eval #:eval ev mod-code ...)
+     (interaction/no-prompt #:eval ev ex-code)))
+
+With the Racket 8.7 release, Typed Racket (TR) includes two languages that
+weaken the run-time behavior of types: Shallow TR and Optional TR.
+Whereas normal TR types (@emph{Deep} types) enforce guarantees that any module
+can depend on, Shallow types enforce only local invariants in typed code, and
+Optional types enforce nothing.
+In return, Shallow and Optional types add less overhead.
+Code often runs faster and simpler than with Deep types.
```

Comment:
You mean when interacting with untyped modules, right? My understanding is that deep types would provide more optimization opportunities within typed modules.

Reply:
The point is that interop overhead is lower, making some things run faster. Probably worth clarifying.